CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
FMA on its own improves precision, but not when taken as a part of a larger expression.

I don't think its that simple. Reading online quickly -- fma _can_ reduce precision, for example when you multiply a complex number by its conjugate (because you end up with something like `a*b-b*a`. You would get 0 normally (and expect to mathematically), but switching to FMA will break the expected cancellation.

Another good example is the way you have to write a LERP using fma's to get the desired properties:
https://fgiesen.wordpress.com/2012/08/15/linear-interpolation-past-present-and-future/

I don't think a compiler can figure it out on its own, and the "old" lerp would start breaking if an FMA was randomly inserted for half the expression.

I would much rather opt-in to FMA explicitly than have it be inserted by the compiler.